### PR TITLE
drop custom compiler patch to provide target

### PR DESCRIPTION
**Pull request by cardoe**
This drops the custom compiler patch to allow us to find the target thereby fixing #138. It also changes how Cargo is built to not use the configure script with make since that is not compatible. It has some improvements to the debug output and lastly provides C++ wrappers to the build process. Lastly it converts the Cargo recipe to one generated by `cargo-bitbake`.

### CODE REVIEW

**Comment by cardoe** [comment]
@Fulkerson I committed some of these patches with you as the author since I got them from your series.

File: recipes-devtools/cargo/cargo_0.16.0.bb [hunk 1]
```diff
@@ -1,85 +1,102 @@
...
+crate://crates.io/winapi-build/0.1.1 \
+crate://crates.io/winapi/0.2.8 \
+crate://crates.io/ws2_32-sys/0.2.1 \
 "
 
-SRCREV_cargo = "6e0c18cccc8b0c06fba8a8d76486f81a792fb420"
 
-S = "${WORKDIR}/git"
 
+# FIXME: update generateme with the real MD5 of the license file
```
Comment (derekstraka) [inline recipes-devtools/cargo/cargo_0.16.0.bb]: Can probably drop this line.  Are the other licenses no longer valid?

Comment (cardoe) [inline recipes-devtools/cargo/cargo_0.16.0.bb]: So `cargo_0.16.0.bb` is purely auto-generated. So yes I can drop that line but its a manual edit until I pull it out of `cargo-bitbake`. The other licenses are valid but not able to be picked up by `cargo-bitbake` for two different reasons. They're listed in `cargo.inc` currently.

Comment (cardoe) [inline recipes-devtools/cargo/cargo_0.16.0.bb]: I'll squash the drop of this line in and I'll also toss on a commit that converts `rustfmt` over to being generated by `cargo-bitbake`.

**Pull request merged**